packages/web/app/src/pages/traces/target-traces-filter.tsx

```diff
@@ -0,0 +1,677 @@
+import {
+  ChangeEventHandler,
+  ComponentPropsWithoutRef,
+  ElementRef,
+  forwardRef,
+  Fragment,
+  InputHTMLAttributes,
+  memo,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { addDays, formatDate, setHours, setMinutes } from 'date-fns';
+import debounce from 'lodash.debounce';
+import {
+  CalendarIcon,
+  CheckIcon,
+  ChevronRightIcon,
+  CircleXIcon,
+  MinusIcon,
+  PlusIcon,
+} from 'lucide-react';
+import type { DateRange } from 'react-day-picker';
+import { Badge } from '@/components/ui/badge';
+import { Button } from '@/components/ui/button';
+import { Calendar } from '@/components/ui/calendar';
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from '@/components/ui/collapsible';
+import { findMatchingPreset, Preset } from '@/components/ui/date-range-picker';
+import { Input } from '@/components/ui/input';
+import { Label } from '@/components/ui/label';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import {
+  SidebarGroup,
+  SidebarGroupContent,
+  SidebarGroupLabel,
+  SidebarMenu,
+  SidebarMenuButton,
+  SidebarSeparator,
+} from '@/components/ui/sidebar';
+import { formatNumber } from '@/lib/hooks';
+import { useResetState } from '@/lib/hooks/use-reset-state';
+import { cn } from '@/lib/utils';
+import * as SliderPrimitive from '@radix-ui/react-slider';
+import * as dateMath from '../../lib/date-math';
+
+interface FilterInputProps extends InputHTMLAttributes<HTMLInputElement> {}
+
+export const FilterInput = forwardRef<HTMLInputElement, FilterInputProps>(
+  ({ className, type, ...props }, ref) => {
+    return (
+      <input
+        type={type}
+        className={cn(
+          'border-input placeholder:text-muted-foreground focus-visible:ring-ring flex h-9 w-full rounded-md border bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium focus-visible:outline-none focus-visible:ring-1 disabled:cursor-not-allowed disabled:opacity-50',
+          className,
+        )}
+        ref={ref}
+        {...props}
+      />
+    );
+  },
+);
+
+export function FilterLocalSearch(props: { value: string; onChange(value: string): void }) {
+  const handleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    props.onChange(e.target.value);
+  }, []);
```

Comment:
```suggestion
  }, [props.onChange]);
```
this should take props.onChange as a dependency